topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -229,11 +215,29 @@ public void handle(Long oldTimerId) {
                     }
                 };
                 periodic.handle(null);
-                promise.future().onComplete(start);
                 LOGGER.info("Started");
             });
     }
 
+    void startWatcher(Promise<Void> start) {
+        Promise<Void> promise = Promise.promise();
+        try {
+            LOGGER.debug("Watching KafkaTopics matching {}", config.get(Config.LABELS).labels());
+
+            Session.this.topicWatch = kubeClient.customResources(CustomResourceDefinitionContext.fromCrd(Crds.kafkaTopic()), KafkaTopic.class, KafkaTopicList.class, DoneableKafkaTopic.class)
+                    .inNamespace(config.get(Config.NAMESPACE)).withLabels(config.get(Config.LABELS).labels()).watch(watcher);
+            LOGGER.debug("Watching setup");
+            // start the HTTP server for healthchecks
+            healthServer = this.startHealthServer();
```

Comment:
We only want to start the health server the first time this is called, not when it's being called to recreate the watch.